keyboards/cassette42/keymaps/default/keymap.c

```diff
@@ -0,0 +1,183 @@
+/* Copyright 2019 monksoffunk
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#ifdef RGBLIGHT_ENABLE
+// Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+rgblight_config_t        RGB_current_config;
+#endif
+
+#include "common/oled_helper.h"
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    RGBRST = SAFE_RANGE,
+    WRTROM,
+};
+
+enum layer_number {
+    _AUDIO = 0,
+    _HUE,
+    _SAT,
+    _VAL,
+    _MODE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    // LAYOUT(LeftEncoder, RightEncoder, LeftSwitch, CenterLeftSwitch, CenterRightSwitch, RightSwitch)
+    [_AUDIO] = LAYOUT(KC_MUTE, KC_ENT, LT(_HUE, KC_MPRV), LT(_SAT, KC_MPLY), LT(_VAL, KC_MNXT), LT(_MODE, KC_SPC)), 
+    [_HUE]   = LAYOUT(RGB_TOG, RGBRST, _______, _______, RGB_HUD, RGB_HUI), 
+    [_SAT]   = LAYOUT(_______, _______, _______, _______, RGB_SAD, RGB_SAI), 
+    [_VAL]   = LAYOUT(_______, _______, RGB_VAD, RGB_VAI, _______, RGB_VAI), 
+    [_MODE]  = LAYOUT(_______, WRTROM, RGB_RMOD, RGB_MOD, RGB_MOD, _______),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case WRTROM:
+#ifdef RGBLIGHT_ENABLE
+            if (record->event.pressed) {
+                eeconfig_update_rgblight(rgblight_config.raw);
+            }
+#endif
+            return false;
+            break;
+
+        case RGBRST:
+#ifdef RGBLIGHT_ENABLE
+            if (record->event.pressed) {
+                eeconfig_update_rgblight_default();
+                rgblight_enable();
+                RGB_current_config.mode = rgblight_config.mode;
+            }
```

Comment:
```suggestion
            if (record->event.pressed) {
                uint8_t temp_mode = rgblight_get_mode();
                eeconfig_update_rgblight_default();
                rgblight_enable();
                rgblight_mode(temp_mode);
            }
```

Reply:
I just want to reset to default mode.

Reply:
Then, just this would be needed.
```suggestion
            if (record->event.pressed) {
                eeconfig_update_rgblight_default();
                rgblight_enable();
            }
```